### PR TITLE
Add downstream build CI job

### PR DESCRIPTION
**Pull request by christophfroehlich**
#200

### CODE REVIEW

File: .github/workflows/binary-build.yml [hunk 1]
```diff
@@ -26,3 +26,17 @@ jobs:
       ros_distro: ${{ matrix.ROS_DISTRO }}
       ros_repo: ${{ matrix.ROS_REPO }}
       ref_for_scheduled_build: master
+  build-downstream:
+    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-industrial-ci-with-cache.yml@test_downstream
```
Comment (christophfroehlich) [inline .github/workflows/binary-build.yml:30]:
```suggestion
    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-industrial-ci-with-cache.yml@master
```

File: .github/workflows/binary-downstream-build.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - master
+
+jobs:
+  build-downstream:
+    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-industrial-ci-with-cache.yml@test_downstream
```
Comment (christophfroehlich) [inline .github/workflows/binary-downstream-build.yml:16]:
```suggestion
    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-industrial-ci-with-cache.yml@master
```

**Comment by christophfroehlich** [comment]
what is your opinion: Not testing and just building everything is fine here?

**Comment by saikishor** [comment]
> what is your opinion: Not testing and just building everything is fine here?

I don't think we have to test anything for now. We just build and that's it and for the testing we have the nightly builds. If we have a use case in future, let's do that

**Review by saikishor — APPROVED** [review]
LGTM

**Comment by fmauch** [comment]
> > what is your opinion: Not testing and just building everything is fine here?
> 
> I don't think we have to test anything for now. We just build and that's it and for the testing we have the nightly builds. If we have a use case in future, let's do that

I would agree in that case.

**Comment by christophfroehlich** [comment]
@mergifyio backport humble

**Pull request merged**